### PR TITLE
KQueue.isAvailable() should return false on MacOS < 10.12

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We currently dont support MacOS < 10.12 so we should return faluse from KQueue.isAvailable() if we are running on an older version.

Modifications:

Check for MacOS version and return false if too old.

Result:

KQueue.isAvailable() will return false if used on an unsupported OSX version.

### CODE REVIEW

**Comment by Scottmitch** [comment]
`transport-native-kqueue` is compatible with 10.11, it just doesn't support linking with 10.11 when built on 10.12 (which is what we release with). This would prevent folks folk using `transport-native-kqueue` if they build their own.

**Comment by normanmaurer** [comment]
@Scottmitch let me come up with a better way.

File: transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueue.java [hunk 1]
```diff
@@ -46,8 +49,29 @@
         if (cause != null) {
             UNAVAILABILITY_CAUSE = cause;
         } else {
-            UNAVAILABILITY_CAUSE = PlatformDependent.hasUnsafe() ? null :
+            cause = PlatformDependent.hasUnsafe() ? null :
                     new IllegalStateException("sun.misc.Unsafe not available");
+            if (cause == null) {
+                if (SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).contains("mac")) {
```
Comment (netkins) [inline transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueue.java]: MAJOR Merge this if statement with the enclosing one. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)

File: transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueue.java [hunk 2]
```diff
@@ -46,8 +49,29 @@
...
+                    String version = SystemPropertyUtil.get("os.version", "");
+                    String[] versionParts = SystemPropertyUtil.get("os.version", "")
+                            .toLowerCase(Locale.US).split("\\.");
+                    if (versionParts.length < 2) {
+                        cause = new IllegalStateException("MacOS version " + version + " not supported");
+                    } else {
+                        try {
+                            int majorVersion = Integer.parseInt(versionParts[0]);
+                            int minorVersion = Integer.parseInt(versionParts[1]);
+                            if (majorVersion < 10 || (majorVersion == 10 && minorVersion < 12)) {
```
Comment (netkins) [inline transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueue.java]: MAJOR Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)

**Comment by Scottmitch** [comment]
I propose closing this PR in favor of https://github.com/netty/netty/pull/6915

**Comment by normanmaurer** [comment]
Thanks @Scottmitch Lets do it...

**Pull request closed without merging**